### PR TITLE
feat(services): centralize playbook discovery in @ansible/services

### PR DESCRIPTION
**Pull request by cidrblock**
## Summary
- Extract playbook discovery into a reusable `discoverPlaybooks()` function in `@ansible/services`
- Uses content-based validation (`parsePlaybook`) to avoid false positives from role files, variable files, and other YAML
- Renames `vitest.config.ts` → `vitest.config.mts` to fix vite 7 ESM compatibility without version pinning

## Changes
- **New**: `packages/services/src/PlaybookDiscovery.ts` — filesystem walker with directory exclusions (`node_modules`, `.git`, `venv`, etc.) and `parsePlaybook()` content validation. Skips role-internal directories (`tasks`, `handlers`, `defaults`, `vars`, `meta`, `templates`, `files`) at recursion time for performance in large workspaces.
- **New**: `packages/services/test/services/PlaybookDiscovery.test.ts` — 13 tests covering valid discovery, exclusion patterns, sorting, edge cases
- **Modified**: `packages/services/src/index.ts` — exports `discoverPlaybooks` and `DiscoveredPlaybook`
- **Modified**: `src/services/PlaybooksService.ts` — delegates to centralized `discoverPlaybooks()` instead of inline discovery logic
- **Renamed**: `vitest.config.ts` → `vitest.config.mts` — forces ESM loading so vite 7 (ESM-only) works with vitest 3.x without version overrides
- **Modified**: `eslint.config.mjs` — updated `allowDefaultProject` to include `.mts` extension

## Test plan
- [x] `npm exec eslint -- .` passes
- [x] `npm exec tsc -- -b` compiles cleanly
- [x] `npm exec vitest -- run` — 678/678 tests pass (42 files)
- [x] New test file covers: valid playbooks, variable files, role internals, hidden dirs, node_modules, venv, collections, nested paths, sorting, non-YAML, empty dirs

### CODE REVIEW

File: packages/services/src/PlaybookDiscovery.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    for (const entry of entries) {
+        const fullPath = path.join(dir, entry.name);
+
+        if (entry.isDirectory()) {
+            if (SKIP_DIRS.has(entry.name) || entry.name.startsWith('.')) {
+                continue;
+            }
+            await walkDir(fullPath, rootDir, results);
+            continue;
+        }
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply packages/services/src/PlaybookDiscovery.ts:85]: Good catch. Refactored walkDir to track `insideRole` state during recursion and skip role-internal directories (tasks, handlers, defaults, vars, meta, templates, files) entirely — avoids unnecessary I/O in large workspaces with many roles. Fixed in 2f0cacd1.
